Unicode problem - Regression??

Hi.

I'm getting text related problems in 2 applications. It worked previously with Haxe 4-rc2 and hxcpp 4.0.8, but it has troubles with Haxe 4-rc3 (or last build) and hxcpp 4.0.19 (or last git version).

**App 1** (OpenFL): it reads a text file to fill the text on screen (it uses i18n and several files for each language).

Result Haxe 4 rc2 + hxcpp 4.0.8:

![image](https://user-images.githubusercontent.com/1063719/61172836-c32bac80-a58a-11e9-8d30-ce8a23e99ab1.png)

Result Haxe 4 rc3 or latest version + hxcpp 4.0.19 or git version (you can see missing letters too):

![image](https://user-images.githubusercontent.com/1063719/61172945-4e597200-a58c-11e9-851d-871a61726291.png)


**App 2** (command line app): It uses a third party C++ library (with externs) to open image files. It was working previosly with Haxe 4 rc2, but I had a problem with `Sys.getEnv("USERPROFILE")`. It returned wrong characters if the username had non-ascii characters. So I saw that bug fixed in Haxe 4 rc3 and the problem was fixed. But now I can't read images from command line. I had problems sending unicode characters from outside, so I used an `encodeUri` method externally and `StringTools.urlDecode` in the haxe side, working in rc2. It isn't working in rc3.

I can replicate the problem with the next code (you need to create a file with the "Test様.txt" name):

```
var uri:String = StringTools.urlDecode("Test%E6%A7%98.txt");
var st:cpp.StdString = cpp.StdString.ofString(uri);
trace(uri, st.toString());
var file:FileInput = File.read(st.toString());
var b = file.readAll();
file.close();
trace(b.length);
```

Result rc2 + 4.0.8:

```
XiaomiMijiaHDRTest01µºÿ.jpg,XiaomiMijiaHDRTest01µºÿ.jpg
5528724
```

Result rc3 + 4.0.19:

```
XiaomiMijiaHDRTest01æ§~.jpg,X
Error: [file_open,X
```

I'm using Windows 10.

I actually can't even compile your example when I have the `ó`. It's fine if I remove it. Latest OpenFL and Lime, rc4 and rc5, hxpp 4.0.52.
```
Error: Main.cpp
./src/Main.cpp(40) : error C2022: '62603' : too big for character
./src/Main.cpp(40) : error C2065: 'u' : undeclared identifier
./src/Main.cpp(40) : error C2143: syntax error : missing ')' before 'string'
./src/Main.cpp(40) : error C2143: syntax error : missing '(' before 'string'
./src/Main.cpp(40) : error C2022: '6113' : too big for character
./src/Main.cpp(40) : error C2022: '32816' : too big for character
./src/Main.cpp(40) : error C2070: 'unknown-type': illegal sizeof operand
./src/Main.cpp(40) : error C2059: syntax error : ')'
```

--- a/include/hx/StdString.h
+++ b/include/hx/StdString.h
@@ -15,8 +15,8 @@ namespace hx
          StdString(const StdString &inS) : std::string(inS) { }
 
          #if (HXCPP_API_LEVEL>1)
-         StdString(const Dynamic &inS) : std::string(inS.mPtr ? inS.mPtr->toString().__s : "null") { }
-         StdString(const String &inS) : std::string(inS.__s , inS.length) { }
+         StdString(const Dynamic &inS) : std::string(inS.mPtr ? inS.mPtr->toString().utf8_str() : "null") { }
+         StdString(const String &inS) : std::string(inS.utf8_str()) { }
          String toString() const { return String(c_str(),size()).dup(); }
          String toString() { return String(c_str(),size()).dup(); }
          operator Dynamic() const { return const_cast<StdString*>(this)->toString(); }

